### ISSUE TITLE
Setting a custom ignore pattern doesn't work more than once

### ISSUE THREAD

**Issue by matthew-dean** [bevry/watchr#81]
Describing the bug is tricky (and there's probably a better way to title it), so bear with me.

I'm using an ignore pattern similar to what's described in issue #50.

So, the first time I set up a watcher, say for ```/Users/path```, it sets an ignore pattern of ```/Users/path/.*/.*``` (I'm omitting regex escapes for clarification). So far so good. It adds the sub-files of that directory, but not the sub-files of any sub-directory.

Now, say I have ```/Users/path/foo``` as a directory. It's been added by the previous watcher but not any of its sub-files. SO, when I go to subsequently watch ```/Users/path/foo``` with a similar ignore pattern, it doesn't work, because watchr rejects ```/foo``` as already being watched, and therefore doesn't watch any sub-files of foo. I could have added those sub-files the first time I set a watcher, but if any of them are directories, I'm just kicking the can down to the next iteration.

In short, because watchr implicitly watches all the paths of a directory, if you exclude them the first time, there's no way to implicitly or explicitly watch them later, unless you were to manually iterate through the list of files and send an array of all paths.

Or is there something I'm missing and some other way to do this?

**Comment by matthew-dean** [comment]
Further update: in my testing, even manually adding files / subdirectories to the first sub-directory, or closing and re-adding the watcher does not work. ```/Users/path/foo``` will not fire any added or deleted events, even though it was "watched" the first time.

**Comment by matthew-dean** [comment]
Looking at the log files, it looks like the ```ignoreCustomPatterns``` is pulling double duty. It prevents sub-directories from being added, but it also prevents events for that subdirectory from ever firing. So even if I later add a watcher to that sub-directory, events for that watcher don't fire because of the initial ignore condition. (I'm seeing "Ignored delete" and the like from the log function).

Maybe a simple option to limit recursion depth (as in #50) would effectively separate these issues? That way, I could explicitly watch certain directories and not their children, and if I later watch their children, they wouldn't be blocked by an ignore condition. The only other solution I can think of is to add a regex "filter" when adding watchers which is separate from the event ignoring condition. However, the first option seems similar and the typical use case.

**Comment by matthew-dean** [comment]
I've now tried several workarounds to disable ignoring after watcher creation, just so events won't be blocked, including:
* setting ignoreCustomPatterns to null after the next() function fires, and/or after the watching event on the watcher instance
* passing ```{ignoreCustomPatterns: null}``` to the setOptions method
* monkey-patching ```isIgnoredPath()``` so that it always returns false for that watcher

I'd rather not fork the library to accomplish this. Any suggestions?

**Comment by balupton** [comment]
v3.0.0 is just released that may fix this problem, as it is a major rewrite, however the https://github.com/bevry/scandirectory package that we use I'll happily replace or accept pull requests, as I'm sure there are better ways these days of doing such things - providing we can support the same ignore configuration options.

Our newer https://github.com/bevry/readdir-cluster may also be a good start on this.

Going to close due to age, but happy to re-open if anyone wants to tackle this.

Sorry for the late response on this, 2014 and 2015 were hard years for me!

**Comment by balupton** [comment]
dupe of https://github.com/bevry/ignorefs/issues/3 which has been resolved downstream in ignorefs and shortly scandirectory, and will be resolved in the next significant release of watchr
